torch/csrc/autograd/variable.cpp

```diff
@@ -87,6 +89,37 @@ std::shared_ptr<Function> VariableImpl::get_grad_accumulator() {
   return result;
 }
 
+VariableViewImpl::VariableViewImpl(Variable base_, at::Tensor data_)
+  : VariableImpl(std::move(data_))
+  , base(std::move(base_))
+  , expected_version(0) {
+  if (!base.defined()) {
+    throw std::runtime_error("base is undefined");
+  }
+  if (base.is_view()) {
+    base = base.base();
+  }
+  is_view = true;
+  version_counter = base.version_counter();
+  expected_version = version_counter.current_version();
+}
+
+std::shared_ptr<Function>& VariableViewImpl::get_grad_fn() {
+  std::lock_guard<std::mutex> lock(grad_accumulator_lock);
```

Comment:
can we rename `grad_accumulator_lock` to `var_lock` or sth like this? It seems unrelated to `grad_accumulator` here